### PR TITLE
do not ignore invalid kwargs in distributions fit method

### LINKED ISSUE #4932: BUG: stats: The `fit` method of the distributions silently ignores invalid keyword arguments.
This issue was noted in this question on stackoverflow: http://stackoverflow.com/questions/30585679/gamma-fit-over-axis-3d-array

The `fit` method silently ignores keyword arguments that it does not use.  It should raise a `TypeError` when given an unknown keyword argument.

Here's a simple `fit` example:

    In [20]: from scipy.stats import gamma

    In [21]: gamma.fit([1,2,3,3,3,4,5,7])
    Out[21]: (3.6899039741659925, 0.049374155400321737, 0.93515411814698823)

The following two calls should raise a `TypeError`.  Instead, the invalid argument is ignored:

    In [22]: gamma.fit([1,2,3,3,3,4,5,7], axis=123)
    Out[22]: (3.6899039741659925, 0.049374155400321737, 0.93515411814698823)

    In [23]: gamma.fit([1,2,3,3,3,4,5,7], flibbity=123)
    Out[23]: (3.6899039741659925, 0.049374155400321737, 0.93515411814698823)

### PR DESCRIPTION
**Pull request by ev-br**
closes gh-4932.

Also fix an unrelated bug where the fit method of gamma and beta distributions did not understand `fa` or `fb` keyword arguments. The issue was that `gamma` and `beta` override the generic `fit` method, so that the generic handling of the keyword args was not used.

And while I'm at it, add an option of specifying shape parameters to fix during fitting as `fix_a`, `fix_b` etc, following the suggestion in  https://github.com/scipy/scipy/pull/4718#issuecomment-93979359. With this PR, 
for example, these all are equivalent for the gamma distribution which has shape parameters `a, b`: `stats.gamma.fit(data, f0=1)`, `stats.gamma.fit(data, fa=1)` and `stats.gamma.fit(data, fix_a=1)`

### CODE REVIEW

**Comment by ev-br** [comment]
CI failure is a timeout.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -419,8 +419,10 @@ def fit(self, data, *args, **kwds):
         # Override rv_continuous.fit, so we can more efficiently handle the
         # case where floc and fscale are given.
 
-        f0 = kwds.get('f0', None)
-        f1 = kwds.get('f1', None)
+        f0 = (kwds.get('f0', None) or kwds.get('fa', None) or
+              kwds.get('fix_a', None))
```
Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: Do these extra names make sense if the shapes are not named ``a, b``?

Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: If they are added, then they should also be documented in the ``fit`` docstring.

Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: This is specific to `beta`, so they are `a, b`.

Comment (rgommers) [inline scipy/stats/_continuous_distns.py]: Ah, missed that. Then this is fine. ``rv_continuous.fit`` docstring still needs a tweak.

Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Good catch --- done.
(OT: adding commits via github web interface actually works, wow)

**Comment by rgommers** [comment]
LGTM, merged. Thanks @ev-br.

**Pull request merged**